### PR TITLE
Add Basic CLI Structure & Start Command

### PR DESCRIPTION
**Pull request by sampoder**

### CODE REVIEW

File: tun/src/unix/linux/mod.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use socket2::{Domain, SockAddr, Socket, Type};
 use std::fs::OpenOptions;
 use std::io::{Error, Write};
 use std::mem;
-use std::net::{Ipv4Addr, Ipv6Addr, SocketAddrV4, SocketAddrV6};
+use std::net::{Ipv4Addr, Ipv6Addr, SocketAddrV4};
 use std::os::fd::RawFd;
 use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd};
```
Comment (Muirrum) [inline tun/src/unix/linux/mod.rs:9]: Not sure why this was changed?

Also: Please [rebase](https://git-rebase.io) your commits to comply with the git lint rules.

**Review by Muirrum — CHANGES_REQUESTED** [review]

File: tun/src/unix/linux/mod.rs (hunk 1 above)
Comment (sampoder) [reply tun/src/unix/linux/mod.rs:9]: Wack. Not sure, why but this can be reversed

File: tun/src/unix/linux/mod.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ use socket2::{Domain, SockAddr, Socket, Type};
 use std::fs::OpenOptions;
 use std::io::{Error, Write};
 use std::mem;
-use std::net::{Ipv4Addr, Ipv6Addr, SocketAddrV4, SocketAddrV6};
+use std::net::{Ipv4Addr, Ipv6Addr, SocketAddrV4};
```
Comment (sampoder) [inline tun/src/unix/linux/mod.rs:7]:
```suggestion
use std::net::{Ipv4Addr, Ipv6Addr, SocketAddrV4, SocketAddrV6};
```

**Comment by sampoder** [comment]
Which command would I use to add a body?

Screenshot 2023-06-06 at 8 57 33 PM

**Comment by Muirrum** [comment]
You should use the `reword` command to edit the commit message. That will keep the file changes intact and only change the message.

**Comment by sampoder** [comment]
awesome! and to add a body do I create a new line or should I just write a
certain amount of characters?

i'll also use reword to remove the trailing punctuation

**Comment by Muirrum** [comment]
To add a body you should add two newlines, like this:

```
This is my commit title!

This is my commit body. There's a line of whitespace separating me from the title.
```

**Comment by sampoder** [comment]
hopefully that fixes things!

**Comment by sampoder** [comment]
oh no :( the linter says the body is too long

**Comment by sampoder** [comment]
hurrah! it works!

**Review by Muirrum — APPROVED** [review]
Excellent! LGTM!

**Pull request merged**